backend/src/main/java/com/ody/mate/controller/MateController.java

```diff
@@ -32,14 +32,11 @@ public ResponseEntity<MeetingSaveResponse> save(
             @AuthMember Member member,
             @RequestBody MateSaveRequest mateSaveRequest
     ) {
-        Long meetingId = InviteCodeGenerator.decode(mateSaveRequest.inviteCode());
-        Meeting meeting = meetingService.findById(meetingId);
-        Mate mate = mateService.save(mateSaveRequest, meeting, member);
-
-        notificationService.saveAndSendDepartureReminder(meeting, mate, member.getDeviceToken());
-
+        Meeting meeting = meetingService.findById(InviteCodeGenerator.decode(mateSaveRequest.inviteCode()));
+        Mate mate = mateSaveRequest.toMate(meeting, member);
+        mateService.save(mate);
+        notificationService.saveEntryAndDepartureNotification(mate);
```

Comment:
[제안]

메서드가 역할 2개를 담당하는 느낌이 들어서 메서드를 분리하는 건 어떨까요? 카키의 의견이 궁금합니다!

```suggestion
        notificationService.saveEntry(mate);
        notificationService.saveDepartureTime(mate);
```

혹은 파라미터를 통해 재사용성을 높여도 괜찮다고 생각합니다!
```suggestion
        notificationService.save(mate, ENTRY);
        notificationSErvice.save(mate, DEPARTURE);
```

Reply:
하나의 트랜젹선으로 묶여야 하는 과정이네요! 조조 의견에 동의합니다.

Reply:
하나의 트랜잭션으로 묶으면서 메서드를 재사용할 수 있게 하고 싶은데 
그렇게 되면 토픽 구독이 필요 없는 입장알림도 구독을 하게 되는 문제가 있게 되더라구요 
아니면 하나의 메서드에서 알림 타입별로 분기 처리를 통해 로직을 수행하면 일부 공통 로직은 제거가 되지만
오히려 가독성이 저해가 되는 느낌을 받았어요 🥲

Reply:
카키 의견에 동의합니다
내부 로직이 완전히 다르니, 하나로 묶기보단 분리하는 게 좋은 것 같아요!